open_spiel/games/sheriff.cc

```diff
@@ -0,0 +1,392 @@
+      return {static_cast<double>(num_illegal_items) * conf.item_value - bribe,
+              bribe};
+    }
+  }
+}
+
+std::unique_ptr<State> SheriffState::Clone() const {
+  return std::make_unique<SheriffState>(*this);
+}
+
+std::string SheriffState::InformationStateString(Player player) const {
+  SPIEL_CHECK_TRUE(player >= 0 && player < NumPlayers());
+
+  std::string infostring = absl::StrCat("T=", MoveNumber(), " ");
+  if (player == kSMUGGLER) {
+    infostring = "num_illegal_items:";
+    if (num_illegal_items_) {
+      absl::StrAppend(&infostring, *num_illegal_items_);
+    } else {
+      absl::StrAppend(&infostring, "none");
+    }
+  }
+
+  SPIEL_CHECK_GE(inspection_feedback_.size() + 1, bribes_.size());
+  SPIEL_CHECK_LE(inspection_feedback_.size(), bribes_.size());
+  for (size_t index = 0; index < bribes_.size(); ++index) {
+    absl::StrAppend(&infostring, "/bribe:", bribes_.at(index));
+
+    if (index < inspection_feedback_.size()) {
+      absl::StrAppend(&infostring,
+                      "/feedback:", inspection_feedback_.at(index));
+    }
+  }
+
+  return infostring;
+}
+
+void SheriffState::UndoAction(Player player, Action action_id) {
+  SPIEL_CHECK_TRUE(!history_.empty() &&
+                   (history_.back() == PlayerAction{player, action_id}));
+  history_.pop_back();
+
+  if (bribes_.size() > 0) {
+    if (bribes_.size() == inspection_feedback_.size()) {
+      // The last action must have been for the sheriff to return feedback about
+      // whether or not the sheriff would inspect the cargo.
+      inspection_feedback_.pop_back();
+    } else {
+      // The last action must have been for the smuggler to offer a new bribe.
+      bribes_.pop_back();
+    }
+  } else {
+    // If there are no bribes yet, then the only possibility is that the game
+    // has just started and the only action so far was for the smuggler to
+    // select the number of illegal items to place into the smuggler's cargo.
+    SPIEL_CHECK_TRUE(num_illegal_items_);
+    num_illegal_items_ = absl::nullopt;
+  }
+}
+
+void SheriffState::DoApplyAction(Action action_id) {
+  SPIEL_CHECK_FALSE(IsTerminal());
+
+  if (!num_illegal_items_) {
+    // The action must represent the selection of the number of illegal items in
+    // the cargo.
+
+    SPIEL_CHECK_EQ(CurrentPlayer(), kSMUGGLER);
+    num_illegal_items_ =
+        sheriff_game_->DeserializeItemPlacementAction(action_id);
+  } else if (bribes_.size() == inspection_feedback_.size()) {
+    // The action must represent a new bribe made by the smuggler.
+    SPIEL_CHECK_EQ(CurrentPlayer(), kSMUGGLER);
+    bribes_.push_back(sheriff_game_->DeserializeBribe(action_id));
+  } else {
+    // The action must represent the inspection feedback returned by the
+    // sheriff.
+    SPIEL_CHECK_EQ(CurrentPlayer(), kSHERIFF);
+    inspection_feedback_.push_back(
+        sheriff_game_->DeserializeInspectionFeedback(action_id));
+  }
+}
+
+// Facts about the game
+const GameType kGameType{
```

Comment:
We normally put these further up closer to the top of the file so people can easily find them, do you need it down here for any reason?

We also usually put this (and the REGISTERS_SPIEL_GAME, I think) in an anonymous namespace.

Reply:
Ah thanks, I had missed it. I will move `kGameType` in the anonymous namespace (together with the registration call. 

By the way, I don't mind applying this fix to Battleship as well in parallel later 🙂